src/main/java/com/atlassian/bitbucket/jenkins/internal/model/BitbucketPullRequest.java

```diff
@@ -26,6 +28,26 @@ public BitbucketPullRequest(
         this.toRef = requireNonNull(toRef, "toRef");
     }
 
+    //Requests don't have to have the same state to be equal
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        BitbucketPullRequest that = (BitbucketPullRequest) o;
+        return id == that.id &&
+               fromRef.equals(that.fromRef) &&
+               toRef.equals(that.toRef);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, state, fromRef, toRef);
```

Comment:
You must remove the state here or you break the contract for `hashCode()` 
>If two objects are equal according to the equals(Object) method, then calling the hashCode method on each of the two objects must produce the same integer result. 

Easiest way out here is to just remove the state from the `haschCode`
(much more around this can be found in the most excellent book Effective Java, Item 8)